src/orion/core/worker/consumer.py

```diff
@@ -100,6 +100,51 @@ def consume(self, trial):
             trial.status = 'broken'
             self.experiment.update_trial(trial, status=trial.status)
 
+    def get_execution_environment(self, trial, results_file='results.log'):
+        """Set a few environment variables to allow users and
+        underlying processes to know if they are running under orion
+
+        Parameters
+        ----------
+        results_file: str
+           file used to store results, this is only used by the legacy protocol
+
+        trial: Trial
+           reference to the trial object that is going to be run
+
+        Note
+        ----
+        This functions define the environment variables described below
+
+        ORION_PROJECT: str
+           name of the project the user is currently working on.
+           it defaults to `orion` is no project name was defined
+
+        ORION_EXPERIMENT: str
+           current experiment id that this trial belong to
+
+        ORION_TRIAL_ID: str
+           current trial id that is currently being ran in this process
+
+        ORION_WORKING_DIRECTORY: str
+           orion current working directory
+
+        ORION_RESULTS_PATH: str
+           orion results file that is read by the legacy protocol to get the results of the trial
+           after a successful run
+
+        """
+        env = dict(os.environ)
+
+        env['ORION_PROJECT'] = 'orion'
+        env['ORION_EXPERIMENT'] = self.experiment.name
```

Comment:
This is not the experiment's id. The id is based on (name, metadata.user), and will now be (name, version, metadata.user).